app/src/main/java/org/mozilla/fenix/IntentReceiverActivity.kt

```diff
@@ -96,6 +101,25 @@ class IntentReceiverActivity : Activity() {
             modeDependentProcessors +
             NewTabShortcutIntentProcessor()
     }
+
+    private fun addReferrerInformation(intent: Intent) {
+        // Pass along referrer information when possible.
+        // Referrer is supported for API>=22.
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP_MR1) {
+            return
+        }
+        // NB: referrer can be spoofed by the calling application. Use with caution.
+        referrer?.let {
+            intent.putExtra(EXTRA_ACTIVITY_REFERRER_PACKAGE, it.host)
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                // Category is supported for API>=26.
+                it.host?.let { host ->
+                    val category = packageManager.getApplicationInfo(host, 0).category
```

Comment:
nit: Thinking we should handle `NameNotFoundException` here for the `getApplicationInfo` call? Not worth crashing over it. I am sure we'll see this and seems fine to just not add the category in this case.